microsoft/PowerToys
# [Powertoys Run] Fixed the "Maximum number of results" setting for the Windows Search Plugin

## Summary of the Pull Request
I know this PR is made before any comments on the bug report, but i'd like to create this before that to see if we can improve Powertoys Run. 

Fixed the "Maximum number of results" setting for the Windows Search Plugin and moved the logic to filter out hidden files to Windows Search (to prevent it from returning no results when all items had the hidden flag.

## References

## PR Checklist
* [x] Applies to #5068

## Detailed Description of the Pull Request / Additional comments
I have removed the tests which did the filtering of the hidden flag in code (since this is in the search api). Any suggestions on how to move / readd this test?

## Validation Steps Performed

- I've created the following folder in my users download folder (with alot of images hidden/ not hidden)  with a similar name. 
![powerlauncherindownloadsfolder](https://user-images.githubusercontent.com/16852398/87824286-43a77c00-c875-11ea-8c62-9ea09894cd9c.png)
- I search for the text "powerlauncher".
- i expect the following to be true.
  - i get no more then 5 search results (the setting i have it on)
  - i get only files with the "shown" in the filename.

Please also note the scrollbar to see howmany results there are left.
Before:
![beforeimage](https://user-images.githubusercontent.com/16852398/87824456-96813380-c875-11ea-907f-56604dc6f04e.png)
After:
![afterimage](https://user-images.githubusercontent.com/16852398/87824522-b7e21f80-c875-11ea-859c-11febec1fc4b.png)

Before i forget, i also changed the setting displayHiddenFiles in the contructor to True and saw hidden files in the lists showing up.

- There's also a unit test to ensure the correct WHERE query is generated.
